modules/basics/src/main/java/com/opengamma/strata/basics/currency/MultiCurrencyAmountArray.java

```diff
@@ -129,7 +131,7 @@ public static MultiCurrencyAmountArray of(int size, IntFunction<MultiCurrencyAmo
    */
   public static MultiCurrencyAmountArray of(Map<Currency, DoubleArray> values) {
     values.values().stream().reduce((a1, a2) -> checkSize(a1, a2));
-    return new MultiCurrencyAmountArray(values);
+    return new MultiCurrencyAmountArray(values.size(), values);
```

Comment:
This is wrong, it's supposed to be the number of currency amounts. This should be the size of one of the values in `values`.